### ISSUE TITLE
DispManX does not work if OpenGL driver is installed

### ISSUE THREAD

**Issue by midwan** [midwan/uae4arm-rpi#7]
Edit: this is an issue with DispManX in general, not specific to the emulator.

There are a few problems if the (experimental) OpenGL driver is installed in the system:
- The emulator GUI does not start up at all if started from the console. The application hangs. It opens up if started with `xinit`.
- The emulation screen does not show at all when started. However, the emulation session is still running, you can return to the GUI with F12 and quit the emulation normally.

Note: The above apply when using the (currently default) approach of DispManX + SDL (v1). Tests showed that the SDL (v1) compile works without those problems, though it's many times slower.

Suggestion: move the emulator to SDL2. It should provide us with a nice speed boost also, since we can use GPU accelerated functions for the display.

**Comment by Fourdee** [comment]
@midwan 

> Suggestion: move the emulator to SDL2. It should provide us with a nice speed boost also, since we can use GPU accelerated functions for the display.

That would be awesome :+1: 
SDL2 dev library on Jessie = https://packages.debian.org/jessie/libsdl2-dev ?

**Comment by midwan** [comment]
@Fourdee Yes, that's the one.
I've started experimenting with it a bit, but it will take me some time to fully understand it and add it as a target platform. The old SDL1 calls need to be changed, but also some of the logic that went with it can be replaced with the modern approach that SDL2 offers.

Regarding this issue however, it's not specific to the emulator but rather to the platform in general. I tried other applications as well, everything that used DispManX to open up a screen failed consistently if the OpenGL driver was installed. If you disable the OpenGL driver, things work again.

The SDL2 solution will work, but then we'll have to start things from within a X-session (which we currently do anyway), since it needs a window manager to work with. DispManX provides the low-level resources to glue SDL/GLES/etc.
